server/src/test/info.test.ts

```diff
@@ -0,0 +1,113 @@
+import * as assert from 'assert';
+import * as path from 'path';
+import * as fs from 'fs/promises';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { Location, Position, Range } from 'vscode-languageserver';
+
+import { compileAqua } from '../validation';
+import { DocumentInfo } from '../info';
+import { pathToUri, tokenToLocation } from '../utils';
+
+/**
+ * Load document from file, compile it and return info
+ * @param relPath path relative to test folder
+ * @returns document, document info after compilation
+ */
+async function openDocument(relPath: string): Promise<[TextDocument, DocumentInfo]> {
+    const absPath = path.join(__dirname, relPath);
+    const uri = pathToUri(absPath);
+
+    const file = await fs.open(absPath);
+    const content = await file.readFile();
+    await file.close();
+
+    const document = TextDocument.create(uri, 'aqua', 0, content.toString());
+    // Compile without imports
+    const [_, info] = await compileAqua({ imports: {} }, document);
+
+    return [document, info];
```

Comment:
Not completely sure, but probably can be simplified using something like this
```suggestion
    const absPath = path.join(__dirname, relPath);
    const content = await fs.readFile(absPath, "utf8");
    const document = TextDocument.create(pathToUri(absPath), 'aqua', 0, content);
    // Compile without imports
    const [_, info] = await compileAqua({ imports: {} }, document);
    return [document, info];
```

Reply:
Why not close a file as soon as we don't need it anymore?

Reply:
in my example I don't even have a file. Do you need a file?